clang/test/Driver/sycl-offload-obj-bundler.c

```diff
@@ -0,0 +1,17 @@
+///
+/// Generate host and target object and bundle them into a single object file.
+///
+
+// REQUIRES: clang-driver
+// REQUIRES: x86-registered-target
+
+/// ###########################################################################
+
+// RUN:   %clang -c -fsycl %s 
```

Comment:
Shouldn't we compile for windows OS?
The title says that fix is for Windows only.
I assume we already have similar tests working for Linux.

BTW, can we add a new test case to the existing test for the bundler: https://github.com/intel/llvm/blob/sycl/clang/test/Driver/clang-offload-bundler.c?

Reply:
Lit test is applicable for both Window and Linux.  
Linux was already working but I don't think there was any lit test.
The current fix is to make it fat object file generation work for Windows.

Reply:
As stated by Alexey, windows specific test in test/Driver/clang-offload-bundler.c should be added.

Reply:
clang-offload-bundler.c requires powerpc-registered target so won't run on windows  
clang-offload-bunderl1.cpp uses linux switch to run
Hence it makes sense to create a new test which is not OS dependent, so it can run on all OS.

Reply:
This might be fine too.
The main point here is that running the whole driver with -fsycl (which will parse the source file three times) just to test bundler component is an overkill.

Reply:
To test I need to create the input files to bundler by running
clang --sycl -c -o %t.bc %s  // create .bc
clang -c -o %t.o %s          // create .o
and then invoke bundler.  Which is what clang driver does under the covers.

Reply:
Why do we need to create an object file like that? Can we just create a fake file with `touch %t.o`? 
If the content of this file is important we can use something like `echo "important stuff" > %t.o`?

Reply:
Currently we don't check  if it is a true obj file.  In the future we will when we wrap the .bc file and embed it into the .obj file

Reply:
We have plenty of tests checking the full stack.
Bundler lit test is a unittest which should validate isolated bandler functionality with as minimal overhead as possible.
If content of the object file is not relevant to testing the change, I suggest avoiding creation of true obj file.